badgecheck/tasks/graph.py

```diff
@@ -21,11 +23,22 @@ def fetch_http_node(state, task_meta):
             return task_result(message='Successfully fetched image from {}'.format(url))
         return task_result(success=False, message="Response could not be interpreted from url {}".format(url))
 
-    # TODO: Add JSON-LD COMPACTION
     actions = [add_task(JSONLD_COMPACT_DATA, data=data, node_id=url)]
     return task_result(message="Successfully fetched JSON data from {}".format(url), actions=actions)
 
 
 def jsonld_compact_data(state, task_meta):
     # TODO: Cache-friendly JSON-LD compaction into the Open Badges context
-    pass
+    input_data = json.loads(task_meta.get('data'))
+
+    node_id = task_meta.get('node_id')
+    options = {'documentLoader': CachableDocumentLoader(cachable=task_meta.get('use_cache', True))}
+
+    result = jsonld.compact(input_data, OPENBADGES_CONTEXT_URI_V2, options=options)
+    actions = [add_node(node_id, data=result)]
```

Comment:
Interesting -- I had `actions = [add_action(ADD_NODE, node_id=node_id, data=result)]`, but I presumed I'd have to add `add_action` then; should we have one, like we have `add_task()`?